File: src/main/java/org/cimsbioko/fragment/AdminSecretFragment.java
package org.cimsbioko.fragment;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.EditText;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;
import org.cimsbioko.R;

public class AdminSecretFragment extends DialogFragment {

    private Listener listener;

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        LayoutInflater inflater = requireActivity().getLayoutInflater();
        View content = inflater.inflate(R.layout.admin_secret_fragment, null);
        return new AlertDialog.Builder(getActivity())
                .setTitle(R.string.admin_access)
                .setView(content)
                .setPositiveButton(android.R.string.ok,
                        (dlg, id) -> {
                            EditText passwordField = content.findViewById(R.id.adminSecretEditText);
                            String secret = passwordField.getText().toString();
                            listener.onAdminSecretDialogOk(secret);
                        })
                .setNegativeButton(android.R.string.cancel,
                        (dlg, id) -> listener.onAdminSecretDialogCancel())
                .create();
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        listener = (Listener) context;
    }

    public interface Listener {
        void onAdminSecretDialogOk(String secret);

        void onAdminSecretDialogCancel();
    }
}

File: src/main/kotlin/org/cimsbioko/fragment/AdminSecretFragment.kt
package org.cimsbioko.fragment

import android.app.Dialog
import android.content.Context
import android.content.DialogInterface
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import org.cimsbioko.R
import org.cimsbioko.databinding.AdminSecretFragmentBinding

class AdminSecretFragment : DialogFragment() {

    private var listener: Listener? = null

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog =
            AdminSecretFragmentBinding.inflate(requireActivity().layoutInflater).let { content ->
                AlertDialog.Builder(requireContext())
                        .setTitle(R.string.admin_access)
                        .setView(content.root)
                        .setPositiveButton(android.R.string.ok) { _: DialogInterface?, _: Int ->
                            listener?.onAdminSecretDialogOk(content.adminSecretEditText.text.toString())
                        }
                        .setNegativeButton(android.R.string.cancel) { _: DialogInterface?, _: Int -> listener?.onAdminSecretDialogCancel() }
                        .create()
            }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        listener = context as Listener
    }

    interface Listener {
        fun onAdminSecretDialogOk(secret: String)
        fun onAdminSecretDialogCancel()
    }
}